plugins/outputs/signalfx/signalfx.go

```diff
@@ -311,10 +317,9 @@ func (s *SignalFx) emitDatapoints() {
 }
 
 func (s *SignalFx) fillAndSendDatapoints(buf []*datapoint.Datapoint) {
+outer:
 	for {
 		select {
-		case <-s.done:
-			return
 		case dp := <-s.dps:
 			buf = append(buf, dp)
 			if len(buf) >= s.BatchSize {
```

Comment:
could break out here too and use the same emit and have the main function re-enter. removes that duplicate code.

Reply:
@mdubbyap I think that risks leaving stuff in the channel on shutdown.  Say we hit our batchsize with things remaining in the channel, if we break and go back to emitdatapoints() and the done channel has been closed, then there's no guarantee that we'll go back to fillAndSendDatapoints().  We might just return and leave things in the dps channel.

Reply:
that's fair.